### PR TITLE
feat: ContainerApp-Added additionalPortMappings logic

### LINKED ISSUE #3062: [AVM Module Issue]: Need additionalPortMappings support for Container app Bicep module
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/app/container-app

### (Optional) Module Version

_No response_

### Description

I need to do additional port mapping for container app but I don't see that property in AVM module. I am not sure why additionalPortMappings ingress property is not added to module.

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by oZakari**
## Description

Adds UDT and property for additional port mappings for ingress.

Closes #3062 

## Pipeline Reference

| Pipeline |
| -------- |
|      [avm.res.app.container-app](https://github.com/oZakari/bicep-registry-modules/actions/workflows/avm.res.app.container-app.yml)    |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/app/container-app/main.bicep [hunk 1]
```diff
@@ -217,6 +220,7 @@ resource containerApp 'Microsoft.App/containerApps@2024-03-01' = {
       ingress: disableIngress
         ? null
         : {
+            additionalPortMappings: !empty(additionalPortMappings) ? additionalPortMappings : null
```
Comment (AlexanderSehr) [inline avm/res/app/container-app/main.bicep:223]:
```suggestion
            additionalPortMappings: additionalPortMappings
```
As it is nullable, the default would anyways be `null`

Comment (oZakari) [reply avm/res/app/container-app/main.bicep:223]: Ah, good point.

**Comment by oZakari** [comment]
Thanks for the review @AlexanderSehr. The vnet test passed (the one I changed), but the max (didn't change) one didn't pass as I had deployed the keyvault to soon after and the previous test so it still considered the name to be in-use.

**Comment by AlexanderSehr** [comment]
> Thanks for the review @AlexanderSehr. The vnet test passed (the one I changed), but the max (didn't change) one didn't pass as I had deployed the keyvault to soon after and the previous test so it still considered the name to be in-use.

Makes sense. This comes down to an Azure Caching issue. I recommend to always wait some time after a previous deployment before trying again, especially for bigger tests where many resources are involved. The good news is that you can then just restart the failed job and don't have to re-run the entire pipeline again. But a bit ... annoying none the less.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**